apps/web/utils/logic/variableItem.ts

```diff
@@ -77,19 +113,191 @@ export const getVariableItemValueObject = (
   return newObjectVariable;
 };
 
-export const answerValuesAreNotEqual = (
-  newAnswerValue: AnswerTypes,
-  currentAnswerValue: AnswerTypes,
-): boolean => {
-  if (isArray(newAnswerValue) && isArray(currentAnswerValue)) {
-    return !newAnswerValue.every((value) => currentAnswerValue.includes(value));
-  } else if (isString(newAnswerValue) && isString(currentAnswerValue)) {
-    return newAnswerValue !== currentAnswerValue;
-  } else if (isObject(newAnswerValue) && isObject(currentAnswerValue)) {
-    return !Object.entries(newAnswerValue).every(
-      ([key, value]) => currentAnswerValue[key] === value,
+export const getVariableItemValueNumber = (
+  variableToSet: VariableToSet,
+  getAnswerToCheckValue: AnswerToCheckValueFn,
+): number => {
+  if (isNumber(variableToSet.variableValue)) {
+    return variableToSet.variableValue;
+  } else if (isArray(variableToSet.variableValue)) {
```

Comment:
```suggestion
const { variableValue } = variableToSet;
  if (isNumber(variableValue)) return variableValue;}

  if (!isArray(variableValue)) {
    throw new Error(
      `getVariableItemValueNumber: variableToSet.variableValue must be a number or an array, got ${typeof variableValue}`,
    );
  }
```

Small thing if you want to add this but we could do the base case type checking first to reduce nesting.
IF you added this change then I'd still add a generic thrown error after the for loop if somehow it makes it past that.

Reply:
Looking at this made me realize we don't even need to pass the whole `variableToSet` object. We just need the `variableValue` property. Refactored for that. Also I tend to like my errors at the bottom for organization and readability so I'd want to leave it there. LMK what you think!

Reply:
Yeah for cases where it's a loop or something that may hit the bottom it's good to have the error case at the bottom but I like having the base cases at the top for organization and readability hahaha But yeah that's fine, not a big difference.